src/index.js

```diff
@@ -6,11 +6,17 @@ import template from 'babel-template';
  */
 const buildHelper = template(`
     function HELPER(cls){
+        var setPrototypeOf = Object.setPrototypeOf || (function setPrototypeOf(a, b) {
+            a.__proto__ = b;
+        });
+        var getPrototypeOf = Object.getPrototypeOf || (function getPrototypeOf(a)) {
+            return a.__proto__;
+        });
         function ExtendableBuiltin(){
             // Not passing "newTarget" because core-js would fall back to non-exotic
             // object creation.
-            var instance = Reflect.construct(cls, Array.from(arguments));
-            Object.setPrototypeOf(instance, Object.getPrototypeOf(this));
+            var instance = new (Function.prototype.bind.apply(cls, arguments))();
```

Comment:
The general concern with approaches like this is that `.bind` is known to be slow. Most things optimize this like statements like https://github.com/zloirock/core-js/blob/master/modules/es6.reflect.construct.js#L29, so larger code like that was what I was trying to avoid.